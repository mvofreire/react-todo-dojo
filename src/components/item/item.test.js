import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { Item } from ".";
import { stub } from "sinon";
import { create } from "react-test-renderer";

describe("Testando Component <Item />", () => {
  const props = {
    titulo: "Teste",
    color: "red",
    onComplete: jest.fn,
    onRemove: jest.fn
  };

  it("Renderiza Item sem erros", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Item {...props} titulo="Teste" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Component <Item/> Chama callback quando clica em Remover", () => {
    const onRemove = stub();

    const wrapper = shallow(<Item {...props} onRemove={onRemove} />);
    wrapper.find("button.todo-remove").simulate("click");
    expect(onRemove.callCount).toEqual(1);
  });

  it("Component <Item/> Chama callback quando clica em Completar TODO", () => {
    const onComplete = stub();
    const wrapper = shallow(<Item {...props} onComplete={onComplete} />);
    wrapper.find("button.todo-complete").simulate("click");
    expect(onComplete.callCount).toEqual(1);
  });

  it("Check Ã© Marcado quando Item esta completo", () => {
    const wrapper = shallow(<Item {...props} complete={true} />);
    const check = wrapper.find("input[type='checkbox']").prop("checked");
    expect(check).toEqual(true);
  });

  it("Valida Snapshot de Item completo", () => {
    const wrapper = create(<Item {...props} complete={true} />);
    expect(wrapper.toJSON()).toMatchSnapshot("completo")
  });

  it("Item Component to match snapshot", () => {
    const wrapper = create(<Item {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
