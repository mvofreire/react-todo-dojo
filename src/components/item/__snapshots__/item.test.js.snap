// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando Component <Item /> Item Component to match snapshot 1`] = `
<div
  className="item-todo"
>
  <input
    className="check-todo-complete"
    disabled={true}
    type="checkbox"
  />
  <span
    className="todo-label"
    style={
      Object {
        "textDecoration": "",
      }
    }
  >
    Teste
  </span>
  <button
    className="todo-complete"
    onClick={[Function]}
  >
    ✔
  </button>
  <button
    className="todo-remove"
    onClick={[Function]}
  >
    ✘
  </button>
</div>
`;

exports[`Testando Component <Item /> Valida Snapshot de Item completo: completo 1`] = `
<div
  className="item-todo"
>
  <input
    checked={true}
    className="check-todo-complete"
    disabled={true}
    type="checkbox"
  />
  <span
    className="todo-label"
    style={
      Object {
        "textDecoration": "line-through",
      }
    }
  >
    Teste
  </span>
  <button
    className="todo-complete"
    disabled={true}
    onClick={[Function]}
  >
    ✔
  </button>
  <button
    className="todo-remove"
    onClick={[Function]}
  >
    ✘
  </button>
</div>
`;
